cmake/CMakeLists.txt

```diff
@@ -1016,27 +1016,18 @@ if (CPUINFO_SUPPORTED)
   endif()
 endif()
 
-# bounds checking behavior.
-# throw instead of calling terminate if there's a bounds checking violation.
-# we make it through via a handler so CUDA does not complain
-# The following -DGSL macros are recognized by gsl-lite along with -Dgsl macros
-# no bounds checking in release build so no perf cost
-# if we enable onnxruntime_DISABLE_EXCEPTIONS, gsl will terminate
-if (onnxruntime_DISABLE_EXCEPTIONS)
-  set(CMAKE_CXX_FLAGS_DEBUG "${CMAKE_CXX_FLAGS_DEBUG} -DGSL_TERMINATE_ON_CONTRACT_VIOLATION")
-else()
-  set(CMAKE_CXX_FLAGS_DEBUG "${CMAKE_CXX_FLAGS_DEBUG} -DGSL_THROW_ON_CONTRACT_VIOLATION")
-endif()
-set(CMAKE_CXX_FLAGS_RELEASE "${CMAKE_CXX_FLAGS_RELEASE} -DGSL_UNENFORCED_ON_CONTRACT_VIOLATION")
-set(CMAKE_CXX_FLAGS_RELWITHDEBINFO "${CMAKE_CXX_FLAGS_RELWITHDEBINFO} -DGSL_UNENFORCED_ON_CONTRACT_VIOLATION")
-set(CMAKE_CXX_FLAGS_MINSIZEREL "${CMAKE_CXX_FLAGS_MINSIZEREL} -DGSL_UNENFORCED_ON_CONTRACT_VIOLATION")
+# GSL
+set(GSL_DIR "${REPO_ROOT}/cmake/external/GSL")
+add_subdirectory(${GSL_DIR} EXCLUDE_FROM_ALL)
```

Comment:
GSL is header only.  So we don't need to use their cmake file. Their cmake files may have things that we don't want. For example, there is a GitHub issue in their repo saying that their cmake file would propagate "-std=c++14" to the targets that link to GSL. But we want C++17 instead. We don't need their cmake target to help us setup any compile flag. Therefore there is no need to use their cmake file.

Reply:
we discussed this offline. this particular issue is only applicable to older versions of CMake (<3.8). ORT has a higher minimum version.
so will just stick to using the GSL CMake target for now where applicable.